### PR TITLE
Move use of __wasm_call_ctors to after module load. NFC.

### PR DESCRIPTION
**Pull request by sbc100**
This avoids adding a pointless function wrapper and uses
the `addOnInit` helper function.

We also don't both with the `if !PTHREAD` check since
pthreads already don't call anything in the `__ATINIT__`
array.

Both these changes save a few bytes in the output.

This change is split out from work I am doing to add
support for TLS + dynamic linking.

### CODE REVIEW

**Comment by sbc100** [comment]
ptal.   This is used by followup changes.

File: src/postamble.js [hunk 1]
```diff
@@ -485,6 +485,14 @@ function exit(status, implicit) {
   quit_(status, new ExitStatus(status));
 }
 
+#if '___wasm_call_ctors' in IMPLEMENTED_FUNCTIONS
+#if !DECLARE_ASM_MODULE_EXPORTS
```
Comment (kripken) [inline src/postamble.js:489]: why is this needed? even with this option we write `__wasm_call_ctors` into the global scope, so always calling it as on line 492 should be ok?

Comment (kripken) [reply src/postamble.js:489]: Oh, is the issue that we have not written it out yet?

Could we move this to this code happens after we write it?

Comment (sbc100) [reply src/postamble.js:489]: With NO_DECLARE_ASM_MODULE_EXPORT we don't write `__wasm_call_ctors` (or any wasm exports) to the global scope statically, they get dynamically injected once the module is loaded.  

If we could be sure this code ran after module instantiation we could avoid this.. perhaps we can find a place where that is true, but here in the global scope the module has not yet been instantiated.

Comment (sbc100) [reply src/postamble.js:489]: Am I misunderstanding what is going on?    This is certainly needed to make `NO_DECLARE_ASM_MODULE_EXPORT` tests pass as it stands.

Comment (kripken) [reply src/postamble.js:489]: I think you are right, now that I follow this.

My concern now is code clarity. At least a comment could help. But I think we can just move this into `receiveInstance` after
```js
#if !DECLARE_ASM_MODULE_EXPORTS
    // If we didn't declare the asm exports as top level enties this function
    // is in charge of programatically exporting them on the global object.
    exportAsmFunctions(exports);
#endif
```
The exports definitely exist after that code.

The only risk is ordering. Maybe other OnInits would have been added meanwhile?

Comment (sbc100) [reply src/postamble.js:489]: I think at addOnIinit does the right thing and puts the thing being added first in the list using `unshift` (I assume we always want it ctors to run first?)

Comment (sbc100) [reply src/postamble.js:489]: I tried that and it didn't work for some reason.. I'll try again to be sure.

**Comment by sbc100** [comment]
Refactored to work like other export accesses in receiveInstance (wasmTable and wasmMemory).

**Review by kripken — APPROVED** [review]
